src/browser-connection/index.js

```diff
@@ -70,11 +73,24 @@ export default class BrowserConnection extends EventEmitter {
     }
 
     close () {
-        this.disconnected = true;
-        this.ready        = false;
+        this.ready = false;
 
         this.browserConnectionGateway.stopServingConnection(this);
         clearTimeout(this.heartbeatTimeout);
+
+        this.closed = true;
+        this.emit('closed');
+    }
+
+    switchToIdle () {
+        this.jobQueue = [];
+
+        this.waitingForIdleTimeout = setTimeout(() => {
```

Comment:
`setTimeout` returns an `object`, so the default value for `this.waitingForIdleTimeout` is `null`